main/management/commands/load_fixtures.py

```diff
@@ -0,0 +1,46 @@
+from django.core.management.base import BaseCommand, CommandError
+from django.core.management import call_command
+from django.conf import settings
+
+import os
+
+def list_fixtures_in_dir(rel_path):
+    files = os.listdir(rel_path)
+
+    def is_fixture(fn):
+        return fn.lower()[-5:] == ".json"
+
+    return [
+        os.path.join(rel_path, fn) for fn in files if is_fixture(fn)
+    ]
+
+
+fixture_dirs = {
+    "proposals": list_fixtures_in_dir("proposals/fixtures/"),
+    "studies": list_fixtures_in_dir("studies/fixtures/"),
+    "tasks": list_fixtures_in_dir("tasks/fixtures/"),
+    "observations": list_fixtures_in_dir("observations/fixtures/"),
+    "main": list_fixtures_in_dir("main/fixtures/"),
+}
+
+
+class Command(BaseCommand):
+    help = "Loads fixtures for development in the correct order."
+
+    def add_arguments(self, parser):
+        parser.add_argument("--force", action="store_true")
+
+    def handle(self, *args, **kwargs):
+        if not settings.DEBUG and not kwargs["force"]:
+            raise CommandError(
+                "Refusing to execute command unless DEBUG = True in settings.py"
+            )
+
+        print("Loading fixtures...")
+
+        for app, locations in fixture_dirs.items():
+            print(f"for {app}")
+            for loc in locations:
+                call_command("loaddata", loc)
+
+        print("Done!")
```

Comment:
This looks good to me. I don't really understand when this would be executed though. I'm assuming when doings tests? I think it looks fine, so I'll approve, but maybe you can explain it a bit later.

Reply:
Django commands get imported and run by manage.py. So this class has a fixed structure with certain arguments, and the Django commands subsystem does the rest.



You can check it out here: https://docs.djangoproject.com/en/5.0/howto/custom-management-commands/

This code would be executed when running  `python manage.py load_fixtures`